tutorial/markdown/aspnet/semantic-kernel/semantic-kernel-tutorial.md

```diff
@@ -0,0 +1,388 @@
+
+### 3. Development Environment
+- **.NET 8.0** or later
+- Visual Studio, VS Code, or JetBrains Rider
+- Basic understanding of C# and vector databases
+
+
+## Setting Up the Environment
+
+### 1. Clone and Navigate
+```bash
+git clone https://github.com/couchbase-examples/couchbase-semantic-kernel-quickstart.git
+cd couchbase-semantic-kernel-quickstart/CouchbaseVectorSearchDemo
+```
+
+### 2. Install Dependencies
+```bash
+dotnet restore
+```
+
+### 3. Configuration Setup
+
+Update `appsettings.Development.json` with your credentials:
+
+```json
+{
+  "OpenAI": {
+    "ApiKey": "your-openai-api-key-here",
+    "EmbeddingModel": "text-embedding-3-small"
+  },
+  "Couchbase": {
+    "ConnectionString": "couchbase://localhost",
+    "Username": "Administrator", 
+    "Password": "your-password",
+    "BucketName": "demo",
+    "ScopeName": "semantic-kernel",
+    "CollectionName": "glossary"
+  }
+}
+```
+
+> **Note**: The `BucketName`, `ScopeName`, and `CollectionName` values can be changed to match your Couchbase setup, but you'll need to update the corresponding code references in the demo application accordingly.
+
+## Understanding the Data Model
+
+The demo uses a `Glossary` class that demonstrates Semantic Kernel's vector store data model. The model uses attributes to define how properties are stored and indexed in the vector database.
+
+For a comprehensive guide on data modeling in Semantic Kernel, refer to [Defining your data model](https://learn.microsoft.com/en-us/semantic-kernel/concepts/vector-store-connectors/defining-your-data-model?pivots=programming-language-csharp) in the official documentation.
+
+### The Glossary Model
+
+```csharp
+internal sealed class Glossary
+{
+    [VectorStoreKey]                    
+    public string Key { get; set; }
+    
+    [VectorStoreData(IsIndexed = true)] 
+    public string Category { get; set; }
+    
+    [VectorStoreData]                   
+    public string Term { get; set; }
+    
+    [VectorStoreData]                   
+    public string Definition { get; set; }
+    
+    [VectorStoreVector(Dimensions: 1536)]
+    public ReadOnlyMemory<float> DefinitionEmbedding { get; set; }
+}
+```
+
+## Step-by-Step Tutorial
+
+### Step 1: Prepare Couchbase
+
+Ensure you have the bucket, scope, and collection ready in Couchbase:
+- **Bucket**: `demo`
+- **Scope**: `semantic-kernel`
+- **Collection**: `glossary`
+
+### Step 2: Data Ingestion and Embedding Generation
+
+This step demonstrates how the connector works with Semantic Kernel's vector store abstractions:
+
+**Getting the Collection** - The demo uses `CouchbaseVectorStore.GetCollection<TKey, TRecord>()` to obtain a collection reference configured for Hyperscale index:
+```csharp
+var vectorStore = new CouchbaseVectorStore(scope);
+var collection = vectorStore.GetCollection<string, Glossary>(
+    "glossary", 
+    new CouchbaseQueryCollectionOptions 
+    {
+        IndexName = "hyperscale_glossary_index",  // Hyperscale index name
```

Comment:
It is not clear to me whether the index needs to be created outside the tutorial or if it will be created as part of the tutorial.
Also, if the index needs to be created before ingestion, it is not currently possible.